File: apps/backend/src/EnterpriseAgentOs.Domain/Interfaces/Agents/IAgentRepository.cs
namespace EnterpriseAgentOs.Domain.Interfaces.Agents;

public interface IAgentRepository
{
    Task<IReadOnlyList<AgentRecord>> ListAsync(CancellationToken ct = default);
    Task<AgentRecord?> GetAsync(Guid id, CancellationToken ct = default);
    Task AddAsync(AgentRecord record, CancellationToken ct = default);
    Task UpdateAsync(AgentRecord record, CancellationToken ct = default);
    Task<bool> SoftDeleteAsync(Guid id, CancellationToken ct = default);
    Task UpdateStatusAsync(Guid id, string status, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.common.valueobjects.AgentStatus;
import com.enterpriseagentos.backendjava.domain.features.agents.models.AgentModel;

public interface IAgentRepository {
    CompletableFuture<List<AgentModel>> listAsync();
    CompletableFuture<AgentModel> getAsync(UUID id);
    CompletableFuture<Void> addAsync(AgentModel model);
    CompletableFuture<Void> updateAsync(AgentModel model);
    CompletableFuture<Boolean> softDeleteAsync(UUID id);
    CompletableFuture<Void> updateStatusAsync(UUID id, AgentStatus status);
    CompletableFuture<List<AgentModel>> listByOwnerAsync(UUID ownerId, boolean includeDeleted);
    CompletableFuture<Void> hardDeleteByOwnerAsync(UUID ownerId);
}
